examples/preprocessing/count_featurizer_transformer.py

```diff
@@ -0,0 +1,122 @@
+"""
+=========================================================
+Using CountFeaturizer to featurize frequencies
+=========================================================
+
+Shows how to use CountFeaturizer to transform some categorical variables
+into a frequency feature. CountFeaturizer can often be used to reduce
+training time, classification time, and classification error.
+"""
+from __future__ import print_function
+import matplotlib.pyplot as plt
+import numpy as np
+import time
+import scipy.sparse as ssp
+from sklearn.preprocessing.data import CountFeaturizer
+from sklearn.preprocessing.data import OneHotEncoder
+from collections import OrderedDict
+from sklearn.datasets import make_classification
+from sklearn.ensemble import RandomForestClassifier
+
+# with reference to auto_examples/ensemble/plot_ensemble_oob
+
+
+RANDOM_STATE = 123
+
+n_datapoints = 1000  # 500
+n_informative = 30  # 15
+n_features = 30  # 25
+n_redundant = 0
+
+# Generate a binary classification dataset.
+X, y = make_classification(n_samples=n_datapoints, n_features=n_features,
+                           n_clusters_per_class=1, n_informative=n_informative,
+                           n_redundant=n_redundant, random_state=RANDOM_STATE)
+
+
+# only make these selected features "categorical"
+discretized_features = [0, 1]
+non_discretized_features = \
+    list(set(range(n_features)) - set(discretized_features))
+
+X_non_discretized = X[:, non_discretized_features]
+
+for feature in discretized_features:
+    X_transform_col = (X[:, feature]).astype(int)
+    # to make all categorical variables non-negative
+    col_min = min(np.amin(X_transform_col), 0)
+    X[:, feature] = X_transform_col - col_min
+
+time_start = time.time()
+
+# X_count is X with an additional feature column 'count'
+cf = CountFeaturizer(inclusion=discretized_features)
+X_count = cf.fit_transform(X)
```

Comment:
I would much rather you execute this preprocessing in a pipeline. You are leaking information from the test set into the training set.